src/transformers/modeling_utils.py

```diff
@@ -3460,14 +3475,18 @@ def _find_mismatched_keys(
                         # The model key doesn't start with `prefix` but `checkpoint_key` does so we remove it.
                         model_key = ".".join(checkpoint_key.split(".")[1:])
 
-                    if (
-                        model_key in model_state_dict
-                        and state_dict[checkpoint_key].shape != model_state_dict[model_key].shape
-                    ):
-                        mismatched_keys.append(
-                            (checkpoint_key, state_dict[checkpoint_key].shape, model_state_dict[model_key].shape)
-                        )
-                        del state_dict[checkpoint_key]
+                    if model_key in model_state_dict:
+                        if (
+                            state_dict[checkpoint_key].shape[-1] == 1
+                            and state_dict[checkpoint_key].numel() * 2 == model_state_dict[model_key].numel()
+                        ):
+                            # such mismatched weights are OK for 4bit quantizations. TODO: use config-based condition
+                            pass
```

Comment:
here you can also enforce this condition with `is_quantized`, or even better with `load_in_4bit` to avoid unexpected behaviour with non-quantized models. Also there seems to be a TODO which is a bit unclear from my perspective, could you please elaborate a bit more 🙏 ?

Reply:
This `load_in_4bit ` was exactly what I was looking for, however it is not passed to `_find_mismatched_keys` or even `_load_pretrained_model()`. An even better alternative would be to check module type - still requires to bring whole model to this function and search for module. I left it as is, edited the comment for clarity.